### PR TITLE
Document `string_deref_patterns` feature

### PR DESCRIPTION
**Pull request by mejrs**
Rendered: 
image

### CODE REVIEW

File: src/doc/unstable-book/src/language-features/string-deref-patterns.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+pub fn is_it_the_answer(value: Value) -> bool {
+    match value {
+        Value::String("42") => true,
+        Value::Number(42) => true,
+        _ => false,
+    }
+}
+```
+
+Without this feature other constructs such as an "if let guard" have to be used.
```
Comment (fmease) [inline src/doc/unstable-book/src/language-features/string-deref-patterns.md:28]: An if-let guard would be `Value::String(s) if let "42" = &s => … which is still unstable (`if_let_guard`) and probably shouldn't be mentioned here. What your snippet contains is an ordinary [match guard](https://doc.rust-lang.org/stable/reference/expressions/match-expr.html#match-guards).

```suggestion
Without this feature other constructs such as match guards have to be used.
```

**Comment by fmease** [comment]
Note that this feature isn't meant to be stabilized as far as I remember. It just exists for demonstration or testing purposes. Hopefully it will be superseded by (general) *deref patterns* in the feature.

File: src/doc/unstable-book/src/language-features/string-deref-patterns.md (hunk 1 above)
Comment (mejrs) [reply src/doc/unstable-book/src/language-features/string-deref-patterns.md]: Oops, thanks. I originally had another example but forget to update the text 😆

**Comment by mejrs** [comment]
> So I don't know if it needs an elaborate explainer. On the other hand, it doesn't hurt to add one.

That's my intention 🙂 . I've found myself explaining this several times, so i figured I should just write up an example and this is a good place for it to live.

**Review by fee1-dead — APPROVED** [review]

**Pull request merged**